### ISSUE TITLE
Crossplatform builds

### ISSUE THREAD

**Issue by danhett** [darrenmothersele/cyril#23]
The single numero uno question i get asked: "is there a Windows/Linux build?" :)

**Comment by darrenmothersele** [comment]
Yes, getting that a lot. Someone at the London Creative Coding meetup was attempting a windows build. I'll check in with them at the next meetup.

**Comment by danhett** [comment]
The likelihood of me helping on a Linux build will increase exponentially when this Macbook finally dies, too!

**Comment by darrenmothersele** [comment]
And I'm definitely getting a new ras pi soon!

**Comment by darrenmothersele** [comment]
I've created a new branch, where I'm working on getting a version compiling using CMake...

https://github.com/cyrilcode/cyril/tree/cmake

VERY early days, and has some OSX stuff hard coded as I'm basing it on this: 
https://github.com/judepereira/44a/blob/master/CMakeLists.txt

I'll update here as I make progress.

**Comment by darrenmothersele** [comment]
As of [this commit](https://github.com/cyrilcode/cyril/commit/3f2173db6948056c5bf778b40c29052242b95ba2) I have got CMake build working on OSX. 
Except it's not responding to keyboard input. There was an issue compiling against glfw3 as the keyboard callback function signature had changed.

**Comment by darrenmothersele** [comment]
The CMake is specific to OSX. It's taken a bit of work to get things working due to way oF links in dependencies. Almost ready to start adapting so it works on Linux too.
Finding lots of things to remove/improve/fix during this process!

**Comment by darrenmothersele** [comment]
yay! working CMake build on OSX as of [this commit](https://github.com/cyrilcode/cyril/commit/a8ba91eccd062ec770ff8559bd75d147abb58dce)

**Comment by darrenmothersele** [comment]
[How to write platform checks with CMake](https://cmake.org/Wiki/CMake_HowToDoPlatformChecks)
    Describes how to implement platform or configure checks with CMake.

**Comment by Bk8** [comment]
Why are you using CMake instead of the ofProject generator to keep it simple?

**Comment by darrenmothersele** [comment]
@Bk8 because ofProject generator creates separate projects for different architectures, right? 
I would prefer we had a single cross-platform build. And CMake allows for that.
And CMake works well with the CLion IDE that I use, which is also cross-platform.

Still, I'll keep this issue open for discussion as happy to hear other ideas...

**Comment by Bk8** [comment]
First your programming language is a awesome!!
But building chains are always a pain in the ass, I'm finding it difficult to build, first I tried the Xcode branch and then the CMake, none of them built.  I was trying to add Syphon support but I failed. I'm sure that I'm missing some configurations, but it would be better in my opinion to make it easier to build. The project will get more attention from the oF community. If you keep the oF Project Generator "standard", I mean all the naming conventions like naming ofApp to the main app; you could just drag and drop the source files under my apps in the library folder  and then generate the project on the platform you want to develop using the oF Project Generator. It may also need to have the flex and bison generated files, but it would be easier to hack.

**Comment by prikhi** [comment]
I got this building on archlinux with the following PKGBUILD & CMakeLists.txt:

https://gist.github.com/prikhi/e22fd332e1c2b5d648e4c56792323e3f

Never used cmake so I might be doing some things wrong...

But when after launching, the editor is blank & unresponsive to my input(can still switch workspaces though). Directly editing the .cy files doesn't update the display. 

Not sure where to go from here.
